src/NServiceBus.Core/Forwarding/ForwardReceivedMessages.cs

```diff
@@ -21,16 +23,22 @@ internal ForwardReceivedMessages()
         /// <param name="context">The feature context</param>
         protected internal override void Setup(FeatureConfigurationContext context)
         {
-            context.MainPipeline.Register<ForwardBehavior.Registration>();
-
             var forwardReceivedMessagesQueue = GetConfiguredForwardMessageQueue(context);
-     
+
             context.Container.ConfigureComponent<ForwardReceivedMessagesToQueueCreator>(DependencyLifecycle.InstancePerCall)
                 .ConfigureProperty(p => p.Enabled, true)
                 .ConfigureProperty(t => t.Address, forwardReceivedMessagesQueue);
 
-            context.Container.ConfigureComponent<ForwardBehavior>(DependencyLifecycle.InstancePerCall)
-                .ConfigureProperty(p => p.ForwardReceivedMessagesTo, forwardReceivedMessagesQueue);
+            context.Pipeline.Register<InvokeForwardingPipelineBehavior.Registration>();
+            context.Pipeline.RegisterConnector<ForwardingToDispatchConnector>("Makes sure that forwarded messages gets dispatched to the transport");
+
+            context.Container.ConfigureComponent(b =>
+            {
+                var pipelinesCollection = context.Settings.Get<PipelineConfiguration>();
+                var pipeline = new PipelineBase<ForwardingContext>(b, context.Settings, pipelinesCollection.MainPipeline);
```

Comment:
I think it would be better if we had explicit pipeline registration method on `FeatureConfigurationContext`, same was as we have for incoming pipelines. We might also split the main pipeline registrations into incoming and outgoing.